go-controller/pkg/ovn/controller/egressfirewall/egressfirewall.go

```diff
@@ -450,50 +586,219 @@ func (oc *DefaultNetworkController) addEgressFirewallRules(ef *egressFirewall, p
+			ops, err = libovsdbops.DeleteACLsFromPortGroupOps(oc.nbClient, ops,
+				clusterPG.Name, acls...)
+			if err != nil {
+				return fmt.Errorf("failed to build cleanup from ClusterPortGroup ops: %w", err)
+			}
+		}
+		_, err = libovsdbops.TransactAndCheck(oc.nbClient, ops)
+		if err != nil {
+			return fmt.Errorf("failed to clean up egress firewall ACLs: %w", err)
+		}
+		return nil
+	})
+	return err
+}
+
+// isEgressFirewallACLEqual checks to see if an existing ACL matches a set of criteria
+func isEgressFirewallACLEqual(acl1, acl2 *nbdb.ACL) bool {
+	if acl1 == nil && acl2 == nil {
+		return true
+	}
+	if acl1 == nil || acl2 == nil {
+		return false
+	}
+	if acl1.Name == nil && acl2.Name == nil {
+		if reflect.DeepEqual(acl1.ExternalIDs, acl2.ExternalIDs) {
+			return true
+		}
+	}
+
+	if acl1.Name == nil || acl2.Name == nil {
+		return false
+	}
+
+	if fmt.Sprintf("%.63s", *acl1.Name) == fmt.Sprintf("%.63s", *acl2.Name) {
+		return true
+	}
+
+	return false
+}
+
 // createEgressFirewallACLOps uses the previously generated elements and creates the
 // acls for all node switches
-func (oc *DefaultNetworkController) createEgressFirewallACLOps(ops []ovsdb.Operation, ruleIdx int, match, action, namespace, pgName string, aclLogging *libovsdbutil.ACLLoggingLevels) ([]ovsdb.Operation, error) {
-	aclIDs := oc.getEgressFirewallACLDbIDs(namespace, ruleIdx)
-	priority := types.EgressFirewallStartPriority - ruleIdx
-	egressFirewallACL := libovsdbutil.BuildACLWithDefaultTier(
-		aclIDs,
-		priority,
-		match,
-		action,
-		aclLogging,
-		// since egressFirewall has direction to-lport, set type to ingress
-		libovsdbutil.LportIngress,
-	)
+func (oc *EFController) createEgressFirewallACLOps(ops []ovsdb.Operation, egressFirewallACL *nbdb.ACL, pgName string) ([]ovsdb.Operation, error) {
 	var err error
 	ops, err = libovsdbops.CreateOrUpdateACLsOps(oc.nbClient, ops, oc.GetSamplingConfig(), egressFirewallACL)
 	if err != nil {
-		return ops, fmt.Errorf("failed to create egressFirewall ACL %v: %v", egressFirewallACL, err)
+		return nil, fmt.Errorf("failed to create egressFirewall ACL %#v: %v", egressFirewallACL, err)
 	}
 
 	ops, err = libovsdbops.AddACLsToPortGroupOps(oc.nbClient, ops, pgName, egressFirewallACL)
 	if err != nil {
-		return ops, fmt.Errorf("failed to add egressFirewall ACL %v to port group %s: %v",
+		return nil, fmt.Errorf("failed to add egressFirewall ACL %#v to port group %s: %v",
 			egressFirewallACL, pgName, err)
 	}
 
 	return ops, nil
 }
 
-func (oc *DefaultNetworkController) deleteEgressFirewallRule(namespace, pgName string, ruleIdx int) error {
+func (oc *EFController) GetEgressFirewallACLDbIDsNoRule(namespace string) *libovsdbops.DbObjectIDs {
+	return libovsdbops.NewDbObjectIDs(libovsdbops.ACLEgressFirewall, defaultControllerOwner,
+		map[libovsdbops.ExternalIDKey]string{
+			libovsdbops.ObjectNameKey: namespace,
+		})
+}
+
+func (oc *EFController) GetEgressFirewallACLDbIDs(namespace string, ruleIdx int) *libovsdbops.DbObjectIDs {
```

Comment:
mayeb add a comment here that we currently reuse the previous default controller as an owner, but it may be upgraded at some point to properly reference the egressfirewall controller?